### PR TITLE
Clean up and add example for C.32 - raw pointers

### PR DESCRIPTION
**Pull request by bgloyer**
Added a simple example for [C.32: If a class has a raw pointer (`T*`), consider whether it might be owning](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rc-dtor-ptr)

Also removed mentioning of owning references.  I think owning referecnce are rare and generally not a problem in legacy code.  
R.3 gives an exception for legacy code to have owing raw pointers but R.4 does not for owning references.  If we want to keep the owing references C.33 should probably be updated to delete owing references.

### CODE REVIEW

File: CppCoreGuidelines.md [hunk 1]
```diff
@@ -4985,24 +4985,32 @@ Here `p` refers to `pp` but does not own it.
...
 ##### Reason
 
 There is a lot of code that is non-specific about ownership.
 
 ##### Example
 
-    ???
+    class LegacyClass
+    {
+        Foo* m_owningPtr;
```
Comment (cubbimew) [inline CppCoreGuidelines.md:4998]: I suspect both `LegacyClass and `m_owningPtr` run afoul of NL.10

**Comment by hsutter** [comment]
Editors call: Owning raw pointers and references are already banned under R.3 and R.4. Sometimes they do get used, so we think owning references should still be mentioned. Otherwise, the PR looks good, if you can please restore the owning references we can approve it.

**Comment by cubbimew** [comment]
How about extending the example to something like
```
foo* owning = new foo; // change to unique_ptr or owner<T*>
bar* observer = &b; // keep
baz& owning_ref = *new baz; // owning reference; banned
xyz& observer_ref = x; // keep
```

**Comment by bgloyer** [comment]
> How about extending the example to something like
> 
> ```
> foo* owning = new foo; // change to unique_ptr or owner<T*>
> bar* observer = &b; // keep
> baz& owning_ref = *new baz; // owning reference; banned
> xyz& observer_ref = x; // keep
> ```

Is it enough just to leave examples for pointers?  It is true that when working with legacy code, ownership of member pointers is often not clear and requires some digging.  Owning member pointers was a common practice in older C++.  That is not true for references.  Owning references have always been a bad idea.  Why would anyone have used an owning ```T&``` instead of just ```T```?  I'm not sure I have ever seen one in the wild.  Even in legacy classes, references should be assumed to be non-owning without second guessing if they are really owned.  There are many more owning pointers than references in legacy code - I like examples that reflect what is common.

**Comment by bgloyer** [comment]
I merge the latest so the diff's are clean now.  Anything more on this one?

**Comment by hsutter** [comment]
Editors call: Thanks!

**Pull request merged**